mulog-kafka/test/com/brunobonacci/mulog/publishers/kafka_test.clj

```diff
@@ -0,0 +1,50 @@
+(ns com.brunobonacci.mulog.publishers.kafka-test
+  (:require [midje.sweet :refer :all]
+            [com.brunobonacci.mulog :as mu]
+            [com.brunobonacci.mulog.publishers.container.kafka-setup :as kafka-setup]
+            [clojure.core.async :as async]
+            [jsonista.core :as json]
+            [org.slf4j.impl.mulog]))
+
+(def ^:private ^:const TOPIC "mulog")
+
+(def ^:private json-mapper (json/object-mapper {:decode-key-fn true}))
+
+(defn- <!!? [chan milliseconds]
+  (let [timeout (async/timeout milliseconds)
+        [value _] (async/alts!! [chan timeout])]
+    value))
+
+(let [stop-console-pub (atom nil)
+      stop-kafka-pub   (atom nil)
+      kafka-container  (atom nil)
+      kafka-consumer   (atom nil)]
+  (with-state-changes [(before :facts
+                               (do
+                                 (reset! stop-console-pub (mu/start-publisher! {:type :console}))
+                                 (reset! kafka-container (kafka-setup/start-container!))
+                                 (reset! stop-kafka-pub (mu/start-publisher! {:type      :kafka
```

Comment:
Not sure we need all these atoms here.

The kafka container can be started only once. Here we start and stop the containers on every fact which makes the tests slow to run if we add more tests.

The others could be in a lexical scope of a `let` and wrapped into a `with-test-cluster` macro to manage the start/stop of publishers and consumers.

what do you think?